Doc/config-api.tex

```diff
@@ -274,325 +131,99 @@ \subsection{ConfigSection Objects}
-
-\var{environ} is an environment variable to use as default value instead
-   of specified value.  If the environment variable exists, it
-   will be used for the default value instead of the specified value.
-
-\var{registry} is a registry key to use as default value instead of
-   specified value.  If the registry key exists, it will be used
-   for the default value instead of the specified value.  A
-   specified environment variable takes precedence over this value.
-   \note{This is not implemented yet.}
-
-\var{name} is a key used to get the option from its corresponding section.
-   You do not need to specify this.  It will be set automatically
-   when you put the option into the \class{ConfigManager} instance.
-
-\var{mandatory} is a flag used to determine if the option itself is
-   required to be present.  The idea of a "mandatory option" is
-   a little strange, but I have seen it done.
-
-\var{source} is a flag used to determine whether the option was set
-   directly in the \class{ConfigManager} instance through Python,
-   by a configuration file/command line option, etc.  You do not need
-   to specify this, it will be set automatically during parsing.
-   This flag should have the value of \var{BUILTIN}, \var{COMMANDLINE},
-   \var{CONFIGFILE}, \var{ENVIRONMENT}, \var{REGISTRY}, or \var{CODE}.
+\begin{classdesc}{Option[T]}{description, options, default}
+  This class represents an option whose value has a ``simple'' type.
+  Specifically, we require \code{T(s)} to correctly converts a string \var{s}
+  to an object of type \class{T}. The type is determined at runtime based on
+  the type of the default value. More complex options should inherit this class.
+
+  From the point of view of the external API, every option item is expected to
+  inherit \class{Option}, and is expected to implement the methods described
+  below correctly. On the other hand, the instance attributes are not
+  guaranteed to exist or take the right value. From the point of view of
+  writing custom \class{Option} subclasses, the instance attributes are set by
+  the default \method{__init__} implementation and, apart from \member{_value},
+  are not modified by any other (default) methods.
 \end{classdesc}
 
-\begin{methoddesc}[GenericOption]{acceptsArgument}{}
-return a boolean indicating whether or not the option accepts an argument on 
-the command-line.  For example, boolean options do not accept an argument.
-\end{methoddesc}
-
-\begin{methoddesc}[GenericOption]{cast}{arg}
-cast the given value to the appropriate type.
-\end{methoddesc}
+\begin{memberdesc}[Option]{description}
+  description of this option to be used in the \code{plastex --help} output.
+\end{memberdesc}
 
-\begin{methoddesc}[GenericOption]{checkValues}{value}
-check \var{value} against all possible valid values for the option.
-If the value is invalid, raise an \exception{InvalidOptionError} exception.
-\end{methoddesc}
+\begin{memberdesc}[Option]{options}
```

Comment:
Maybe this name is not great. Something mentioning command line (or cmd_line) may be clearer.

Reply:
How about ConfigOption?